File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    
    const dfs = (index,currentSum) => {
        if (index === numbers.length) {
            if(currentSum === target) {
                answer ++;
            }
            return;
        }
        dfs(index + 1, currentSum + numbers[index]);
        dfs(index + 1, currentSum - numbers[index]);
    }
    
    dfs(0,0);
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
answer = 0

def dfs (numbers,num,index,target) :
    global answer
    if index == len(numbers) :
        if num == target :
            answer += 1
        else :
            return 0 
    else : 
        dfs(numbers, num + numbers[index], index+1,target)
        dfs(numbers, num - numbers[index], index+1,target)
    
def solution(numbers, target):
    global answer
    dfs(numbers,0,0,target)
    
    return answer